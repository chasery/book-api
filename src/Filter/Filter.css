.Filter + .Filter {
  margin-left: 1rem;
}

.Filter label {
  margin-right: 0.5rem;
  font-size: 1.125rem;
}
