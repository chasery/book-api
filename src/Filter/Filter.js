import React from "react";
import "./Filter.css";

class Filter extends React.Component {
  renderOptions(options) {
    return options.map((option, i) => {
      return (
        <option key={i} value={option}>
          {option}
        </option>
      );
    });
  }

  render() {
    const { stringName, name, options, handleInputChange, value } = this.props;

    return (
      <div className="Filter">
        <label htmlFor={name}>{stringName}:</label>
        <select
          name={name}
          id={name}
          onChange={handleInputChange}
          value={value}
        >
          {this.renderOptions(options)}
        </select>
      </div>
    );
  }
}

export default Filter;
