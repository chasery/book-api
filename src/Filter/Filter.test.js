import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Filter from "./Filter";

describe("Filter component", () => {
  it("renders without error", () => {
    const wrapper = shallow(<Filter />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
