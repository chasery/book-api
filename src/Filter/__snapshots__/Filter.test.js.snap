// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component renders without error 1`] = `
<div
  className="Filter"
>
  Filter
</div>
`;
